azkaban-common/src/main/java/azkaban/project/AzkabanProjectLoader.java

```diff
@@ -209,8 +212,13 @@ private void persistProject(final Project project, final FlowLoader loader, fina
         this.projectLoader.uploadProjectProperties(project, directoryFlowLoader.getPropsList());
 
       } else if (loader instanceof DirectoryYamlFlowLoader) {
-        // Todo jamiesjc: upload yaml file to DB as a blob
-
+        final File[] flowFiles = projectDir.listFiles(new SuffixFilter(Constants.FLOW_FILE_SUFFIX));
+        for (final File file : flowFiles) {
+          final int newFlowVersion = this.projectLoader
+              .getLatestFlowVersion(project.getId(), newProjectVersion, file.getName()) + 1;
+          this.projectLoader
```

Comment:
as discussed, maybe we should persist the name of the uploader as well.

Reply:
The name of uploader is already persisted in project_versions DB table. See azkaban.project.ProjectLoader#changeProjectVersion.